src/python/pants/backend/go/goals/tailor_test.py

```diff
@@ -117,6 +117,37 @@ def test_find_go_package_targets(rule_runner: RuleRunner) -> None:
     )
 
 
+def test_cgo_sources(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(
+        {
+            "foo/go.mod": "",
+            "foo/main.go": "",
+            "foo/native.c": "",
+            "foo/another.c": "",
+            "foo/native.h": "",
+            "foo/assembly.s": "",
+        }
```

Comment:
It would be great to add a folder like `c_only`. We want to make sure that 1+ `.go` file must be present for a target to be added

Reply:
Done.